src/sdk_version.rs

```diff
@@ -0,0 +1,87 @@
+use std::fmt;
+use serde::{Serialize, Serializer};
+
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum SDKVersion {
+    Oreo,
+    Nougat,
+    Marshmallow,
+    Lollipop,
+    KitKat,
+    JellyBean,
+    IceCreamSandwich,
+    Honeycomb,
+    Gingerbread,
+    Froyo,
+    Eclair,
+    Donut,
+    Cupcake,
+    Old,
+    Unknown,
+}
+
+use self::SDKVersion::*;
+
+impl From<u32> for SDKVersion {
+    fn from(version: u32) -> Self {
+        match version {
+            26 => Oreo,
+            25 | 24 => Nougat,
+            23 => Marshmallow,
+            22 | 21 => Lollipop,
+            19 => KitKat,
+            18 | 17 | 16 => JellyBean,
+            15 | 14 => IceCreamSandwich,
+            13 | 12 | 11 => Honeycomb,
+            10 | 9 => Gingerbread,
+            8 => Froyo,
+            7 | 6 | 5 => Eclair,
+            4 => Donut,
+            3 => Cupcake,
+            2 | 1 => Old,
+            _ => Unknown,
+        }
+    }
+}
+
+impl From<SDKVersion> for u32 {
```

Comment:
I'm not completely sure if there's a better way to do this in Rust. On the other hand, one SDK version can have multiple API numbers, but not the other way around.

Reply:
Well, one way is to instead of using `SDKVersion` as a type, you could use `ApiNumber` as a type and multiple API versions return the same string.

Not sure why we would need this conversion though, the number comes with the data we parse.